src/cargo/util/toml/mod.rs

```diff
@@ -2561,6 +2592,17 @@ impl<P: ResolveToPath + Clone> DetailedTomlDependency<P> {
     fn update_optional(&mut self, optional: Option<bool>) {
         self.optional = optional;
     }
+
+    fn resolve_path(&mut self, root_path: &Path, package_root: &Path, config: &Config) {
+        if let Some(from_root) = &self.path {
+            let resolved = from_root.resolve(config);
+            if let Some(diff_path) = diff_paths(root_path.join(resolved), package_root) {
```

Comment:
Does `diff_paths` gracefully handle diffing `/foo/bar/../baz` with `/foo/bee`?  Do we need to call `cargo_utils::paths::normlize_path` after doing the `join`?

I remember having to be particular about that in cargo-add but I don't remember if it was related to `diff_paths` or other stuff

(maybe another reason for a shared function, see below)

Reply:
I don't know if it handles that well or not. I have a feeling that we will need to call `cargo_utils::paths::normlize_path` as well. I'll change it to this.